runtime_scan/pkg/scanner/creds/basic.go

```diff
@@ -54,37 +58,34 @@ func (u *BasicRegCred) ShouldAdd() bool {
 	return *u.isSecretExists
 }
 
-// Add The scanner is using docker config json that contains the username and the password required to pull the image.
-// We need to do the following:
-// 1. Create a volume that holds the `BasicRegCredSecretName` data
-// 2. Mount the volume into each container to a specific path (`BasicVolumeMountPath`/`DockerConfigFileName`)
-// 3. Set `DOCKER_CONFIG` to point to the directory that contains the config.json.
+// Add the BasicRegCred image pull secret to the job by mounting it as a volume
+// and configuring the IMAGE_PULL_SECRET_PATH environment variable.
 func (u *BasicRegCred) Add(job *batchv1.Job) {
+	// Move the job to the configured namespace
 	job.Namespace = u.secretNamespace
+
+	// Mount the image pull secret as a folder into a known location in the job
+	volumeName := fmt.Sprintf("image-pull-secret-%s", BasicRegCredSecretName)
 	job.Spec.Template.Spec.Volumes = append(job.Spec.Template.Spec.Volumes, corev1.Volume{
-		Name: BasicVolumeName,
+		Name: volumeName,
 		VolumeSource: corev1.VolumeSource{
 			Secret: &corev1.SecretVolumeSource{
 				SecretName: BasicRegCredSecretName,
-				Items: []corev1.KeyToPath{
-					{
-						Key:  corev1.DockerConfigJsonKey,
-						Path: DockerConfigFileName,
-					},
-				},
 			},
 		},
 	})
 	for i := range job.Spec.Template.Spec.Containers {
 		container := &job.Spec.Template.Spec.Containers[i]
 		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
-			Name:      BasicVolumeName,
+			Name:      volumeName,
 			ReadOnly:  true,
-			MountPath: BasicVolumeMountPath,
-		})
-		container.Env = append(container.Env, corev1.EnvVar{
-			Name:  DockerConfigEnvVar,
-			Value: BasicVolumeMountPath,
+			MountPath: fmt.Sprintf("/opt/kubeclarity-pull-secrets/%s", BasicRegCredSecretName),
 		})
 	}
+
+	// Configure the IMAGE_PULL_SECRET_PATH to the location where we mounted the image pull secret
+	for i := range job.Spec.Template.Spec.Containers {
```

Comment:
Why do we need another for loop here and doing this in the previous one?

Reply:
Great catch, I had copied this from the job_management logic but there is no need for a double loop